File: App_Company/lib/features/booking_reports/domain/entities/booking_report.dart
import 'package:equatable/equatable.dart';

class BookingReport extends Equatable {
  const BookingReport({
    required this.id,
    required this.bookingId,
    required this.customerId,
    required this.customerName,
    required this.companyId,
    required this.companyName,
    this.workerId,
    this.workerName,
    required this.reason,
    this.description,
    required this.status,
    required this.statusName,
    this.adminResolutionNotes,
    this.resolvedByAdminId,
    this.resolvedByAdminName,
    this.resolvedAt,
    required this.createdAt,
    this.updatedAt,
    required this.bookingStatus,
    required this.bookingStatusName,
  });

  final int id;
  final int bookingId;
  final int customerId;
  final String customerName;
  final int companyId;
  final String companyName;
  final int? workerId;
  final String? workerName;
  final String reason;
  final String? description;
  final int status;
  final String statusName;
  final String? adminResolutionNotes;
  final int? resolvedByAdminId;
  final String? resolvedByAdminName;
  final DateTime? resolvedAt;
  final DateTime createdAt;
  final DateTime? updatedAt;
  final int bookingStatus;
  final String bookingStatusName;

  bool get isOpen => status == 0;
  bool get isActive => status == 0 || status == 1;
  bool get isTerminal => status == 2 || status == 3;

  @override
  List<Object?> get props => [
        id,
        bookingId,
        customerId,
        customerName,
        companyId,
        companyName,
        workerId,
        workerName,
        reason,
        description,
        status,
        statusName,
        adminResolutionNotes,
        resolvedByAdminId,
        resolvedByAdminName,
        resolvedAt,
        createdAt,
        updatedAt,
        bookingStatus,
        bookingStatusName,
      ];
}

class BookingReportFilters extends Equatable {
  const BookingReportFilters({
    this.status,
    this.bookingId,
    this.customerId,
    this.workerId,
    this.fromDate,
    this.toDate,
  });

  final int? status;
  final int? bookingId;
  final int? customerId;
  final int? workerId;
  final DateTime? fromDate;
  final DateTime? toDate;

  BookingReportFilters copyWith({
    int? status,
    int? bookingId,
    int? customerId,
    int? workerId,
    DateTime? fromDate,
    DateTime? toDate,
    bool clearStatus = false,
  }) {
    return BookingReportFilters(
      status: clearStatus ? null : (status ?? this.status),
      bookingId: bookingId ?? this.bookingId,
      customerId: customerId ?? this.customerId,
      workerId: workerId ?? this.workerId,
      fromDate: fromDate ?? this.fromDate,
      toDate: toDate ?? this.toDate,
    );
  }

  @override
  List<Object?> get props => [
        status,
        bookingId,
        customerId,
        workerId,
        fromDate,
        toDate,
      ];
}

File: Admin_Web/src/types/booking-report.ts
export const BookingReportStatus = {
  Open: 0,
  InReview: 1,
  Resolved: 2,
  Rejected: 3,
} as const;

export type BookingReportStatusValue =
  (typeof BookingReportStatus)[keyof typeof BookingReportStatus];

export type AdminBookingReportStatusValue = Exclude<BookingReportStatusValue, 0>;

export interface BookingReport {
  id: number;
  bookingId: number;
  customerId: number;
  customerName: string;
  companyId: number;
  companyName: string;
  workerId?: number;
  workerName?: string;
  reason: string;
  description?: string;
  status: BookingReportStatusValue;
  statusName: string;
  adminResolutionNotes?: string | null;
  resolvedByAdminId?: number | null;
  resolvedByAdminName?: string | null;
  resolvedAt?: string | null;
  createdAt: string;
  updatedAt?: string | null;
  bookingStatus: number;
  bookingStatusName: string;
}

export interface BookingReportFilters {
  status?: BookingReportStatusValue;
  bookingId?: number;
  customerId?: number;
  companyId?: number;
  workerId?: number;
  fromDate?: string;
  toDate?: string;
  page?: number;
  pageSize?: number;
}

export interface UpdateBookingReportStatusPayload {
  status: AdminBookingReportStatusValue;
  adminResolutionNotes?: string;
}

export const BOOKING_REPORT_STATUS_LABELS: Record<BookingReportStatusValue, string> = {
  [BookingReportStatus.Open]: 'مفتوح',
  [BookingReportStatus.InReview]: 'قيد المراجعة',
  [BookingReportStatus.Resolved]: 'تم الحل',
  [BookingReportStatus.Rejected]: 'مرفوض',
};

export const BOOKING_REPORT_STATUS_COLORS: Record<BookingReportStatusValue, string> = {
  [BookingReportStatus.Open]: 'bg-orange-100 text-orange-800',
  [BookingReportStatus.InReview]: 'bg-blue-100 text-blue-800',
  [BookingReportStatus.Resolved]: 'bg-green-100 text-green-800',
  [BookingReportStatus.Rejected]: 'bg-red-100 text-red-800',
};

export const ADMIN_BOOKING_REPORT_STATUS_OPTIONS: {
  value: AdminBookingReportStatusValue;
  label: string;
}[] = [
  { value: BookingReportStatus.InReview, label: BOOKING_REPORT_STATUS_LABELS[1] },
  { value: BookingReportStatus.Resolved, label: BOOKING_REPORT_STATUS_LABELS[2] },
  { value: BookingReportStatus.Rejected, label: BOOKING_REPORT_STATUS_LABELS[3] },
];

export const ADMIN_RESOLUTION_NOTES_MAX = 1000;
